packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -917,33 +924,129 @@ class GoTrueClient {
 
   /// Recover session from stringified [Session].
   Future<AuthResponse> recoverSession(String jsonStr) async {
-    final session = Session.fromJson(json.decode(jsonStr));
-    if (session == null) {
-      await signOut();
-      throw notifyException(AuthException('Current session is missing data.'));
-    }
-
-    if (session.isExpired) {
-      if (_autoRefreshToken && session.refreshToken != null) {
-        return await _callRefreshToken(
-          refreshToken: session.refreshToken,
-          accessToken: session.accessToken,
+    try {
+      final session = Session.fromJson(json.decode(jsonStr));
+      if (session == null) {
+        await signOut();
+        throw notifyException(
+          AuthException('Current session is missing data.'),
         );
+      }
+
+      if (session.isExpired) {
+        final refreshToken = session.refreshToken;
+        if (_autoRefreshToken && refreshToken != null) {
+          return await _callRefreshToken(refreshToken);
+        } else {
+          await signOut();
+          throw notifyException(AuthException('Session expired.'));
+        }
       } else {
-        await signOut();
-        throw notifyException(AuthException('Session expired.'));
+        final shouldEmitEvent = _currentSession == null ||
+            _currentSession?.user.id != session.user.id;
+        _saveSession(session);
+
+        if (shouldEmitEvent) {
+          notifyAllSubscribers(AuthChangeEvent.tokenRefreshed);
+        }
+
+        return AuthResponse(session: session);
       }
-    } else {
-      final shouldEmitEvent = _currentSession == null ||
-          _currentSession?.user.id != session.user.id;
-      _saveSession(session);
+    } catch (error) {
+      notifyException(error);
+      rethrow;
+    }
+  }
+
+  /// Starts an auto-refresh process in the background. Close to the time of expiration a process is started to
+  /// refresh the session. If refreshing fails it will be retried for as long as necessary.
+  ///
+  /// If you set the `autoRefreshToken` to `true`, you don't need to call this function, it will be called for you.
```

Comment:
This is not really true. Only when using the `supabase_flutter` package this method is called, nowhere else. Additionally, `autoRefreshToken` should be respected when calling `startAutoRefresh` in `supabase_flutter` as well.

Reply:
Great catch! Pushed a fix.

Reply:
I think the auto refresh should be started when only using the supabase or gotrue  package as well. Not only when using the flutter package

Reply:
Makes sense. Let me change the code right now.

Reply:
@Vinzent03 Updated it to call the startAutoRefresh from within gotrue.